### ISSUE TITLE
Save Button does not Function

### ISSUE THREAD

**Issue by sunraybrad** [factus10/fm-htmleditor#1]
I am having difficulty moving this into another solution. The Save button is not firing. In debug mode the save button fires from the demo, but not from my new solution.
I am seeing the content and rendered HTML fine, I can copy+paste the code view into a field fine, but the Save button is just not firing the script.
Any ideas?

**Comment by factus10** [comment]
I suspect you need to enable the permissions for the callback.

1. Select File / Manage / Security from the FileMaker menus.
2. Click "Advanced Settings" (in FM18).
2. Click on the "Extended Privileges Tab", then double-click the "fmurlscript" keyword.
3. Check "on" for "Full Access".

**Comment by sunraybrad** [comment]
Thanks David, while that was an issue, it did not fix the problem. I did further troubleshooting and it seems to be tied to the length of the content field. So it works with relatively short text but fails on longer pieces of text. Is this related to how long the URL can be, hence the passing of the content code to the callback? Does that make sense? I have not narrowed it down to a specific length value yet.

**Comment by factus10** [comment]
Are you on a Windows machine?

**Comment by sunraybrad** [comment]
Yes. I found this
[https://www.seedcode.com/filemaker-webviewer-javascript-calculation-engine/](url)
Working thru how to implement. Basically using the clipboard to bypass the IE (Windows) character limit. Not sure if I have the experience to pull that off :(

**Comment by factus10** [comment]
Yep, found the same thing. I think the path forward is through clipboard.js. I'll look into it :)

**Comment by sunraybrad** [comment]
I got it to work using the clipboard per the SeedCode article noted above. The new HTMLeditor::html value was modified here:

/* Prep the URL to use for the hand-off to FM. */
**var mydata = $('#summernote').summernote('code');**
var fullURL = '[[SAVEURL]]'; // **param removed**
**window.clipboardData.setData( 'Text' , mydata );**
/* Send the data to FM. */
window.location = fullURL;
return false;

Then the Save HTML script modified to omit the $editedContent var and instead Paste the clipBoard into a global field that is used to set the content.

![sunray_saveHTML](https://user-images.githubusercontent.com/24783714/77332212-249a4280-6cf8-11ea-9206-671096040297.jpg)

**Comment by factus10** [comment]
Cool! I am going to continue to look into integrating clipboard.js since your code is IE specific.

**Comment by ksp-ux** [comment]
> I got it to work using the clipboard per the SeedCode article noted above. The new HTMLeditor::html value was modified here:
> 
> /* Prep the URL to use for the hand-off to FM. _/ **var mydata = $('#summernote').summernote('code');** var fullURL = '[[SAVEURL]]'; // **param removed** **window.clipboardData.setData( 'Text' , mydata );** /_ Send the data to FM. */
> window.location = fullURL;
> return false;
> 
> Then the Save HTML script modified to omit the $editedContent var and instead Paste the clipBoard into a global field that is used to set the content.
> 
> ![sunray_saveHTML](https://user-images.githubusercontent.com/24783714/77332212-249a4280-6cf8-11ea-9206-671096040297.jpg)

Thanks Sunraybrad! I got it working after copying your changes

**Comment by factus10** [comment]
Thank y'all. I'll fix main code with this solution.

**Comment by carlstarus** [comment]
Hi factus10,

Thanks for the sample file. I do see I seem to have a missing function in the script though. I have no idea which function it is. Please check the attached screenshot. Thanks in advance.
![function-missing](https://user-images.githubusercontent.com/25885283/119959657-7e556700-bfa4-11eb-930e-78429d464886.png)

**Comment by factus10** [comment]
Carl,

Sorry about that. I was using a function from another plugin and forgot about it. I've changed it to use InsertFromURL, which does the trick.

Download the new version and let me know what you think.
